packages/api/src/common/dapps/caviarnine/getCaviarnineResourcePoolPositions.ts

```diff
@@ -0,0 +1,159 @@
+import { Effect } from "effect";
+import BigNumber from "bignumber.js";
+
+import { GetFungibleBalanceService } from "../../gateway/getFungibleBalance";
+
+import { CaviarNineConstants } from "./constants";
+import type { AtLedgerState } from "../../gateway/schemas";
+
+import { GetResourcePoolUnitsService } from "../../resource-pool/getResourcePoolUnits";
+
+export class InvalidResourcePoolError extends Error {
+  readonly _tag = "InvalidResourcePoolError";
+  constructor(error: unknown) {
+    super(
+      `Invalid resource pool error: ${error instanceof Error ? error.message : error}`
+    );
+  }
+}
+
+import type { ShapeLiquidityAsset } from "./getShapeLiquidityAssets";
+
+export type GetCaviarnineResourcePoolPositionsOutput = {
+  pool: (typeof CaviarNineConstants.simplePools)[keyof typeof CaviarNineConstants.simplePools];
+  result: {
+    address: string;
+    items: ShapeLiquidityAsset[];
+  }[];
+}[];
+
+type AccountAddress = string;
+
+export class GetCaviarnineResourcePoolPositionsService extends Effect.Service<GetCaviarnineResourcePoolPositionsService>()(
+  "GetCaviarnineResourcePoolPositionsService",
+  {
+    effect: Effect.gen(function* () {
+      const getFungibleBalanceService = yield* GetFungibleBalanceService;
+      const getResourcePoolUnitsService = yield* GetResourcePoolUnitsService;
+
+      return {
+        getCaviarnineResourcePoolPositions: Effect.fn(
```

Comment:
call the method `run`, `getCaviarnineResourcePoolPositions` is inferred from the service name